protos/backend_service.proto

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+syntax = "proto3";
+
+package durabletask.protos.backend.v1;
+
+option csharp_namespace = "Microsoft.DurableTask.Protobuf";
+option java_package = "com.microsoft.durabletask.implementation.protobuf";
+option go_package = "/internal/protos";
+
+import "orchestrator_service.proto";
+
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/empty.proto";
+
+// gRPC service used by Durable Task Framework (DTFx) backend implementations.
+// The RPCs in this service are used by DTFx clients to manage orchestrations.
+service ClientBackendService {
+    // Creates a new orchestration instance.
+    rpc CreateOrchestrationInstance (CreateInstanceRequest) returns (CreateInstanceResponse);
```

Comment:
Note messages' names don't match the RPC name, should be `CreateOrchestrationInstanceRequest` and `CreateOrchestrationInstanceResponse`

Reply:
I'm reusing the existing `CreateInstanceRequest` message defined in orchestration_service.proto. I can't change those messages, so could either rename these RPCs or create duplicate messages with a different name.

Reply:
I would maybe duplicate the messages, since that could allow us to make changes in the future?

Note that even if messages are different (different names and even different field names), if they have fields of the same type, at the same index, they are compatible on the wire. May be hard to rely on that, but it's an option too

Reply:
A reason not to duplicate the message is to ensure consistency between the two APIs, which is one of my goals. If one RPC needs a new field, it's likely that the other RPC also needs that field. It also allows us to reuse existing validation logic across message types.